From 11fe723c28029f35c989e34f37f6b94bc0432c09 Mon Sep 17 00:00:00 2001
From: Yang Xun <yangxun@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:06 +0800
Subject: [PATCH 069/150] dt-bindings: Add a new binding for Phytium display
 engine

This patch documents the DT binding for Phytium display engine.

Signed-off-by: Yang Xun <yangxun@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I4ad406074df5193034c7b7acec8eb9eb5d207517
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/gpu/phytium,dc.yaml   | 53 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 54 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/gpu/phytium,dc.yaml

diff --git a/Documentation/devicetree/bindings/gpu/phytium,dc.yaml b/Documentation/devicetree/bindings/gpu/phytium,dc.yaml
new file mode 100644
index 000000000000..5cea3a283fa0
--- /dev/null
+++ b/Documentation/devicetree/bindings/gpu/phytium,dc.yaml
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/gpu/phytium,dc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium Display Engine
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,dc
+
+  reg:
+    minItems: 1
+    maxItems: 2
+    description: |
+     Physical base address of the registers and length of memory mapped region,
+     and optional physical base address and length of reserved system memory.
+
+  interrupts:
+    maxItems: 1
+
+  pipe_mask:
+    $ref: '/schemas/types.yaml#/definitions/uint32'
+    description: specify which pipe is enabled, each bit corresponds to a pipe.
+
+  edp_mask:
+    $ref: '/schemas/types.yaml#/definitions/uint32'
+    description: specify which pipe is edp port, each bit corresponds to a pipe (0:dp, 1:edp).
+
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - pipe_mask
+  - edp_mask
+
+examples:
+  - |
+    /memreserve/ 0xf4000000 0x4000000; // (optional)
+
+    dc0@32000000 {
+      compatible = "phytium,dc";
+      reg = <0x0 0x32000000 0x0 0x8000>,
+            <0x0 0xf4000000 0x0 0x4000000>; // (optional)
+      interrupts = <GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>;
+      pipe_mask = 0x3;
+      edp_mask = 0x0;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 178676f7dfcb..c49e14bd230b 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2532,6 +2532,7 @@ S:	Maintained
 W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml
+F:	Documentation/devicetree/bindings/gpu/phytium,dc.yaml
 F:	Documentation/devicetree/bindings/hwlock/phytium,hwspinlock.yaml
 F:	Documentation/devicetree/bindings/hwmon/phytium,tacho.yaml
 F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
-- 
2.47.0

